api/src/main/java/io/strimzi/api/kafka/model/KafkaJmxOptionsAuthenticationSimple.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Configures the Kafka Brokers JMX port with username and password protected.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaJmxOptionsAuthenticationSimple extends KafkaJmxOptionsAuthentication {
+    private static final long serialVersionUID = 1L;
+
+    public static final String TYPE_SIMPLE = "simple";
+    private Boolean passwordProtected;
+
+    @Description("Must be `" + TYPE_SIMPLE + "`")
+    @Override
+    public String getType() {
+        return TYPE_SIMPLE;
+    }
+
+    @Description("Whether the Kafka brokers has password protection.")
+    @JsonProperty(value = "password")
+    public Boolean getPasswordProtected() {
```

Comment:
I don't think we need this field anymore because the type identified this, or?

Reply:
I think that we should keep it this way because the next extension to simple auth would be to add a presenting a certificate. I think other expansions on`KafkaJmxOptionsAuthentication` would be then to add other types of authentication.

Reply:
So, it the password need the be enabled or disabeld in this field and ancryption will have its own field - what does the simple authntication type mean? I would expect that `type: simple` would mean password in which case the other field should not be needed, or?

Reply:
Maybe simple is not the right name for the class. Password protection and putting a certificate are basic things that one can use to secure the port on start up. Maybe a better name for the class would be `basic`?

Reply:
I'm fine with that. But it is still not clear to me what does it mean if you have to specify the password in another field. So shouldn't it be `type: password`?

Reply:
So what would the CR for an open JMX port look like: What I envisioned was that a user would have:
```
spec:
  kafka:
    # ...
    jmxOptions:
      authentication:
        type: "simple"
        password: true
    # ...
```

Are you saying the CR for a secure port should look like: 
```
spec:
  kafka:
    # ...
    jmxOptions:
      authentication:
        type: "password"
    # ...
```

and for an open port it should look like:


```
spec:
  kafka:
    # ...
    jmxOptions:
    # ...
```

Reply:
Yeah, that is basically how we do it in other places:

* No authentication:

```yaml
jmxOptions: {}
```

* Password authentication:

```yaml
    jmxOptions:
      authentication:
        type: password
```

* Other authentication (e.g. TLS client certificates):

```yaml
    jmxOptions:
      authentication:
        type: tls
```

That would follow the same logic which we use for example for the authentication on Kafka listeners.

Reply:
alright will make the changes now :)